app/org/sagebionetworks/bridge/cache/CacheProvider.java

```diff
@@ -207,7 +209,31 @@ public void removeString(String cacheKey) {
             throw new BridgeServiceException(e);
         }
     }
-
+    
+    public EmailVerificationStatus getEmailVerificationStatus(String email) {
+        try {
+            String cacheKey = RedisKey.STUDY_EMAIL_STATUS.getRedisKey(email);
+            String statusString = jedisOps.get(cacheKey);
+            return (statusString == null) ? null : EmailVerificationStatus.fromSesVerificationStatus(statusString);
+        } catch (Throwable e) {
+            promptToStartRedisIfLocal(e);
+            throw new BridgeServiceException(e);
+        }
+    }
+    
+    public void setEmailVerificationStatus(String email, EmailVerificationStatus status) {
+        try {
+            String cacheKey = RedisKey.STUDY_EMAIL_STATUS.getRedisKey(email);
+            String result = jedisOps.setex(cacheKey, BridgeConstants.BRIDGE_STUDY_EMAIL_STATUS_IN_SECONDS, status.name());
+            if (!"OK".equals(result)) {
+                throw new BridgeServiceException("View storage error");
+            }
+        } catch (Throwable e) {
+            promptToStartRedisIfLocal(e);
+            throw new BridgeServiceException(e);
+        }
+    }
```

Comment:
It looks like email verification calls are fairly cheap. It's probably not worth building cache logic for this.